components/style/gecko/wrapper.rs

```diff
@@ -661,7 +661,8 @@ impl<'le> ::selectors::Element for GeckoElement<'le> {
             NonTSPseudoClass::MozTableBorderNonzero |
             NonTSPseudoClass::MozBrowserFrame => unsafe {
                 Gecko_MatchesElement(pseudo_class.to_gecko_pseudoclasstype().unwrap(), self.0)
-            }
+            },
+            NonTSPseudoClass::MozAny(ref selectors) => matches(&selectors.0, self, None),
```

Comment:
> Can you verify that something like http://searchfox.org/mozilla-central/source/layout/reftests/css-selectors/state-dependent-in-any.html (perhaps tweaking manually with :hover or :active, or something we actually implement instead of :valid) works?

Selectors like `:-moz-any(:hover)` do not seem to update correctly on state changes; I'm not certain why, since as you said `AFFECTED_BY_STATE` is always set.

Also, things like `:-moz-any(:first-child)` are valid and should set `HAS_EDGE_CHILD_SELECTOR` and `AFFECTED_BY_CHILD_INDEX`, so I think we do need to pass the relations and flags here.